src/main/java/uk/ac/cam/cl/dtg/isaac/api/IsaacController.java

```diff
@@ -360,29 +374,30 @@ public final Response getDocumentByPath(@Context final Request request, @Context
 
             ByteArrayOutputStream fileContent;
             String mimeType;
-            switch (Files.getFileExtension(path).toLowerCase()) {
-                case "pdf":
-                    mimeType = "application/pdf";
-                    break;
-
-                default:
-                    // if it is an unknown type return an error
-                    SegueErrorResponse error = new SegueErrorResponse(Status.BAD_REQUEST, "Invalid file type requested");
-                    return error.toResponse(getCacheControl(NUMBER_SECONDS_IN_ONE_DAY, false), etag);
+            if (Files.getFileExtension(path).equalsIgnoreCase("pdf")) {
+                mimeType = "application/pdf";
+            } else {
+                // if it is an unknown type return an error
+                SegueErrorResponse error = new SegueErrorResponse(Status.BAD_REQUEST, "Invalid file type requested");
+                return error.toResponse(getCacheControl(NUMBER_SECONDS_IN_ONE_DAY, false), etag);
             }
 
             fileContent = this.contentManager.getFileBytes(path);
             if (null == fileContent) {
                 String refererHeader = httpServletRequest.getHeader("Referer");
-                SegueErrorResponse error = new SegueErrorResponse(Status.NOT_FOUND, "Unable to locate the file: " + path);
+                SegueErrorResponse error = new SegueErrorResponse(
+                        Status.NOT_FOUND, "Unable to locate the file: " + path);
                 log.warn(String.format("Unable to locate the file: (%s). Referer: (%s)", path, refererHeader));
                 return error.toResponse(getCacheControl(NUMBER_SECONDS_IN_TEN_MINUTES, false), etag);
             }
 
+            RegisteredUserDTO currentlyLoggedInUser = userManager.getCurrentRegisteredUser(httpServletRequest);
```

Comment:
It makes sense to check this early on like we did before, otherwise we're doing unnecessary work for a user who isn't logged in.